### DISCUSSION TITLE
Live Stream/Video Player failure

### DISCUSSION THREAD

**Discussion by marksteven** [gatsbyjs/gatsby#36532] [Help]
### Preliminary Checks

- [X] This issue is not a duplicate. Before opening a new issue, please search existing issues: https://github.com/gatsbyjs/gatsby/issues
- [X] This issue is not a question, feature request, RFC, or anything other than a bug report directly related to Gatsby. Please post those things in GitHub Discussions: https://github.com/gatsbyjs/gatsby/discussions

### Description

Simple Gatsby minimal site plus any React Video player either live stream or file -mp4 work perfect in Develop but causes Build to fail.

### Reproduction Link
https://github.com/marksteven/GatsbyVideoPLayerFailure.git
or
https://codesandbox.io/s/hardcore-galileo-qkmq84?file=/src/pages/index.js

### Steps to Reproduce

1. gatsby new VideoPlayerExampleBuildBug https://github.com/gatsbyjs/gatsby-starter-minimal
2. yarn install
3. gatsby develop - works fine no errors
4. gatsby build - no errors
5. yarn add clappr - A React video player version 0.3.13 was installed
6.  Replace Index with simple video player page using clappr
7. Gatsby develop - Works perfect no errors
8. gatsby build - fails 
9. Errors 

 8 |  else
   9 |          root["Clappr"] = factory();
> 10 | })(window, function() {
     |  ^
  11 | return /******/ (function(modules) { // webpackBootstrap
  12 | /******/         // The module cache
  13 | /******/         var installedModules = {};

  WebpackError: ReferenceError: window is not defined
  
  - clappr.js:10 
    [gatsby-starter-minimal]/[clappr]/dist/clappr.js:10:2
  
  - bootstrap:19 
    gatsby-starter-minimal/webpack/bootstrap:19:1
  
  - bootstrap:19 
    gatsby-starter-minimal/webpack/bootstrap:19:1
  
  - static-entry.js:214 
    gatsby-starter-minimal/.cache/static-entry.js:214:27
  
  - utils.js:154 
    [gatsby-starter-minimal]/[@gatsbyjs]/reach-router/lib/utils.js:154:1
...

### Expected Result

expected result was build would work just like develop.
Have now tried using simple gatsby site with 4 different react video players to date
All with similar results on build yet development works fine
Posts online talk about adding gatsby-node to alleviate the window error so have added this 
however it did not seem to work
exports.onCreateWebpackConfig = ({ stage, loaders, actions }) => {
    if (stage === "build-html" || stage === "develop-html") {
      actions.setWebpackConfig({
        module: {
          rules: [
            {
              test: /clappr/,
              use: loaders.null(),
            },
          ],
        },
      })
    }
  }

### Actual Result

develop works
build fails
8 |  else
   9 |          root["Clappr"] = factory();
> 10 | })(window, function() {
     |  ^
  11 | return /******/ (function(modules) { // webpackBootstrap
  12 | /******/         // The module cache
  13 | /******/         var installedModules = {};

  WebpackError: ReferenceError: window is not defined
  
  - clappr.js:10 
    [gatsby-starter-minimal]/[clappr]/dist/clappr.js:10:2
  
  - bootstrap:19 
    gatsby-starter-minimal/webpack/bootstrap:19:1
  
  - bootstrap:19 
    gatsby-starter-minimal/webpack/bootstrap:19:1
  
  - static-entry.js:214 
    gatsby-starter-minimal/.cache/static-entry.js:214:27
  
  - utils.js:154 
    [gatsby-starter-minimal]/[@gatsbyjs]/reach-router/lib/utils.js:154:1

### Environment

```shell
System:
    OS: macOS 11.6.8
    CPU: (8) x64 Intel(R) Core(TM) i7-4960HQ CPU @ 2.60GHz
    Shell: 5.8 - /bin/zsh
  Binaries:
    Node: 18.7.0 - /usr/local/bin/node
    Yarn: 1.22.19 - ~/.yarn/bin/yarn
    npm: 8.15.0 - /usr/local/bin/npm
  Languages:
    Python: 2.7.16 - /usr/bin/python
  Browsers:
    Chrome: 104.0.5112.101
    Firefox: 101.0.1
    Safari: 15.6
  npmPackages:
    gatsby: ^4.21.1 => 4.21.1
  npmGlobalPackages:
    gatsby-cli: 4.20.0
```

### Config Flags

no flags

**Comment by bobbydp** [comment]
you are running into this (official gatsby documentation):
https://www.gatsbyjs.com/docs/debugging-html-builds/#how-to-check-if-window-is-defined
